Apollo.podspec

```diff
@@ -34,6 +34,14 @@ Pod::Spec.new do |s|
     ss.dependency 'Apollo/Core'
     ss.dependency 'SQLite.swift', '~>0.12.2'
   end
+  
+  # Cache persistance storage mechanism extended with encryption mechanism
+  s.subspec 'SQLiteCipher' do |ss|
+    ss.source_files = 'Sources/ApolloSQLite/*.swift'
+    ss.dependency 'Apollo/Core'
+    ss.dependency 'SQLCipher', '~>4.2.0'
+    ss.dependency 'SQLite.swift/SQLCipher', '> 0.12.0'
+  end
```

Comment:
We're not actually using anything from SQLite cipher inside our library, so I don't think this is necessary (and would add a couple of dependencies that could cause conflicts.

Reply:
The main problem is - it seems to be impossible to use it together with Apollo/SQLite. 
`SQLite.swift/SQLCipher` - requires only this particular SQLite dependency.
Mixing it together with`ss.dependency 'SQLite.swift', '~>0.12.2'` results in not working SQLCipher layer. It is not going to cause any error, crash or compilation warning. Just DB is not going to encrypt at all.

That is why I prepared separate dependency.  
Please check this comment and the answer:
https://github.com/stephencelis/SQLite.swift/issues/570#issuecomment-268530061
This is the exact situation. 

I was already trying to use 
`pod Apollo/SQLite `
`pod SQLCipher`
together with this init
```
    self.shouldVacuumOnClear = shouldVacuumOnClear
    self.db = db
    try self.createTableIfNeeded()
  }
```

and DB is not encrypting. 

Any thoughts how to resolve?

Reply:
Oof, looking at [the SQLitePodspec](https://github.com/stephencelis/SQLite.swift/blob/master/SQLite.swift.podspec#L70) it looks like there's some additional `xcconfig`s that get set up in order to make that work. Looking at the [`README` for FMDB, another iOS SQLite lib](https://github.com/ccgus/fmdb#cocoapods), I see a similar caveat about how to set up the podspec. 

Looking at the tutorial [on SQLCipher](https://www.zetetic.net/sqlcipher/ios-tutorial/), there's a fairly blunt caveat about only using their SQLite 3: 
 
<img width="1152" alt="Screen Shot 2020-04-17 at 3 27 05 PM" src="https://user-images.githubusercontent.com/1976498/79611379-1dc8da80-80c0-11ea-98e9-2e83e1fb507d.png">

Neither SQLite.swift nor FMDB support using this through anything other than CocoaPods, which is a big old red flag for me in terms of maintaining support on the Apollo side, which would be necessary if we were to add this to the library. 

Honestly, at this point, it feels like we're pretzeling ourselves to make this work in a way that just doesn't make sense for the overwhelming majority of users. This is a major reason why `NormalizedCache` is a protocol: so that you can implement your own version of it if the one we provide is insufficient.